<?xml version="1.0" encoding="utf-8"?>
<launch>
	<arg name="object_class" default="apple/Apple" />
	<arg name="object_frame" default="apple/apple" />
	<arg name="camera_frame" default="camera_color_frame" />
	<arg name="fx" default="604.7211303710938" />
	<arg name="fy" default="603.2431640625" />
	<arg name="sub_depth" default="/camera/aligned_depth_to_color/image_raw" />
	<arg name="sub_boxes" default="/darknet_ros/bounding_boxes" />
	<arg name="sub_image" default="/camera/color/image_raw" />
	<arg name="pub_image" default="/object_located/image_raw" />

	<arg name="pub_object_in_base_link" default="/object_in_base_link" />
	<arg name="pub_object_in_base_camera" default="/object_in_base_camera" />


	<node name="object_located" pkg="vision" type="object_located" output="screen">
		<param name="object_class" value="$(arg object_class)" />
		<param name="object_frame" value="$(arg object_frame)" />
		<param name="camera_frame" value="$(arg camera_frame)" />
		<param name="fx" value="$(arg fx)" />
		<param name="fy" value="$(arg fy)" />
		<remap from="/camera/color/image_raw"  to="$(arg sub_image)" />
		<remap from="/object_located/image_raw"  to="$(arg pub_image)" />
		<remap from="/camera/aligned_depth_to_color/image_raw"  to="$(arg sub_depth)" />
		<remap from="/darknet_ros/bounding_boxes"  to="$(arg sub_boxes)" />
	</node>
	<node name="pub_object_in_coordinate" pkg="vision" type="pub_object_in_coordinate" output="screen">
		<remap from="/object_in_base_link"  to="$(arg pub_object_in_base_link)" />
		<remap from="/object_in_base_camera"  to="$(arg pub_object_in_base_camera)" />
	</node>
</launch>

